src/order/chain.lean

```diff
@@ -219,3 +230,47 @@ let ⟨h₁, H⟩ := chain_closure_max_chain.is_chain.super_chain_succ_chain h i
   H.ne (chain_closure_max_chain.succ_fixpoint_iff.mpr rfl).symm
 
 end chain
+
+/-! ### Flags -/
+
+/-- The type of flags, aka maximal chains, of an order. -/
+structure flag (α : Type*) [has_le α] :=
+(carrier : set α)
+(chain' : is_chain (≤) carrier)
```

Comment:
Is there precedent to using `is_chain (≤)` instead of `is_chain (<)`? The definition of `is_chain r` reads as 
```lean
∀ x y, x ≠ y → r x y ∨ r y x
```
so in a way, `≤` causes some redundancy.

Reply:
This is on purpose. It makes `flag.chain'` easier to prove, then API lemmas get you back to the stricter statement.